### PR TITLE
feat[pages]: Add unit tests for `wrangler pages publish`

### PR DESCRIPTION
**Pull request by CarmenPopoviciu**

### CODE REVIEW

File: packages/wrangler/src/__tests__/pages.test.ts [hunk 1]
```diff
@@ -1000,6 +1001,779 @@ describe("pages", () => {
...
+			});
+
+			setMockResponse(
+				"/accounts/:accountId/pages/projects/foo/deployments",
+				async ([_url, accountId], init) => {
+					assertLater(() => {
+						expect(accountId).toEqual("some-account-id");
+						expect(init.method).toEqual("POST");
+						const body = init.body as FormData;
+						const manifest = JSON.parse(body.get("manifest") as string);
```
Comment (GregBrimble) [inline packages/wrangler/src/__tests__/pages.test.ts:1173]: Would be nice to check for `_worker.js` here too.

File: packages/wrangler/src/__tests__/pages.test.ts [hunk 2]
```diff
@@ -1000,6 +1001,779 @@ describe("pages", () => {
...
+			"✨ Success! Uploaded 1 files (TIMINGS)
+
+			✨ Uploading _worker.js
+			✨ Deployment complete! Take a peek over at https://abcxyz.foo.pages.dev/"
+			`);
+
+			expect(std.err).toMatchInlineSnapshot('""');
+		});
+
+		it("should ignore _worker.js if the /functions directory exists", async () => {
```
[the remark this answers is not in the document]
Comment (GregBrimble) [reply packages/wrangler/src/__tests__/pages.test.ts:1199]: I'd maybe just remove this test then, and let's file a GitHub Issue to track the work of flipping this precedence around.

File: packages/wrangler/src/__tests__/pages.test.ts [hunk 3]
```diff
@@ -1000,6 +1001,779 @@ describe("pages", () => {
...
+			);
+
+			setMockResponse(
+				"/accounts/:accountId/pages/projects/foo/deployments",
+				async ([_url, accountId], init) => {
+					assertLater(() => {
+						expect(accountId).toEqual("some-account-id");
+						expect(init.method).toEqual("POST");
+						const body = init.body as FormData;
+						const manifest = JSON.parse(body.get("manifest") as string);
```
Comment (GregBrimble) [inline packages/wrangler/src/__tests__/pages.test.ts:1645]: Once again, let's confirm `_routes.json` here.

File: packages/wrangler/src/__tests__/pages.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (CarmenPopoviciu) [reply packages/wrangler/src/__tests__/pages.test.ts:1199]: fair enough. I'll do that

File: packages/wrangler/src/__tests__/pages.test.ts [hunk 4]
```diff
@@ -1077,16 +1077,47 @@ describe("pages", () => {
...
+						// make sure this is all we uploaded
+						expect([...body.keys()]).toEqual(["manifest", "_worker.js", "_routes.json"]);
+
 						expect(manifest).toMatchInlineSnapshot(`
 				                          Object {
 				                            "/README.md": "13a03eaf24ae98378acd36ea00f77f2f",
 				                          }
 			                      `);
+
+						// the contents of the generated `_worker.js` file is pretty massive, so I don't
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/__tests__/pages.test.ts:1102]: @GregBrimble pls let me know if you have other thoughts on this

Comment (GregBrimble) [reply packages/wrangler/src/__tests__/pages.test.ts:1102]: Yeah, what you have there is perfect. Thank you!

**Review by GregBrimble — APPROVED** [review]
Approved (once you run Prettier) :)

**Pull request merged**